### PR TITLE
Messaging input toolbar

### PR DESCRIPTION
**Pull request by MTTHWBSH**
Figured I would start getting some of the Pitnerest related work merged as we're coming up on closing that out as well as integrating some of this stuff into Stylist app (I can hold off on merging until we're ready, or just plan on not updating this dependency for the apps until they're ready).
- Adds toolbar view below input view
- Accepts array of "actions" that provide convenience callbacks for tapping

### CODE REVIEW

File: Source/MessagingTextInputbar.m [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+@property (nonatomic, strong) NSLayoutConstraint *editorContentViewHC;
+
+@end
+
+@implementation MessagingTextInputbar
+
+@synthesize customToolbarView = _customToolbarView;
+
+- (CGFloat)appropriateHeight {
+    CGFloat height = super.appropriateHeight;
```
Comment (amyschmidt) [inline Source/MessagingTextInputbar.m:38]: curious about the syntax, is there a certain time and place you put the { on the same line whereas the other ones have it on the next line

Comment (MTTHWBSH) [reply Source/MessagingTextInputbar.m:38]: Nah that should be dropped to the next line based on our Obj-C styleguide, will do.

Was there another question about the actual syntax or just bracket placement?

Comment (amyschmidt) [reply Source/MessagingTextInputbar.m:38]: just about that bracket

**Comment by mccarron** [comment]
LGTM but I'd rather more members of our team look over it since I paired with you on it.

**Comment by MTTHWBSH** [comment]
@jdivittorio3 would you mind taking a look at this? Preparing to merge a few Pinterest dependencies in the next day or two.

File: Source/ToolbarAction.m [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+//
+//
+
+#import "ToolbarAction.h"
+
+@implementation ToolbarAction
+
+- (void)setBackgroundImage:(NSString *)imageName
+{
+    [self.imageView setContentMode: UIViewContentModeScaleAspectFill];
```
Comment (joediv) [inline Source/ToolbarAction.m:15]: Double checking to make sure you want this to be aspect fill? Images will most likely be the correct proportion anyway.

**Review by joediv — APPROVED** [review]
LGTM 🚀

File: Source/ToolbarAction.m (hunk 2 above)
Comment (MTTHWBSH) [reply Source/ToolbarAction.m:15]: Yeah, it should always fit perfectly based on constraints - more so just a double check to make sure that the provided image is both taking up as much space in the container as possible and respecting its aspect ratio.

**Pull request merged**